### PR TITLE
Update "Enterprise Version Compatibility" and "Upgrade Airflow" docs with 1.10.15, Helm Chart

### PR DESCRIPTION
**Pull request by paolaperaza**
Hey folks - updating a few more files to

- Reflect the recent addition of 1.10.15 + 2.0.1
- Reflect the addition of 0.18.7 of our Airflow chart

Astronomer v0.16, to my knowledge, is not compatible with AC 1.10.15 + 2.0.1, so I did not touch some v0.16 files.

@vishwas-astro @jwitz You two mind taking a look at this?

### CODE REVIEW

File: enterprise/v0.16/09_resources/03_version-compatibility-reference.md [hunk 1]
```diff
@@ -12,12 +12,12 @@ It's worth noting that while the tables below reference the minimum compatible v
...
+| v0.16                | 1.16, 1.17, 1.18 | 3    | 0.12, 0.13.5| 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14                        | 3.6, 3.7, 3.8  | 0.16           |
+| v0.23                | 1.16, 1.17, 1.18 | 3    | 0.13.5      | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.1 | 3.6, 3.7, 3.8  | 0.23           |
 
-> **Note:** Astronomer v0.16.9+ is required to run Astronomer Certified 1.10.12, and Astronomer v0.16.15+ is required to run Astronomer Certified 1.10.14. For instructions on how to upgrade, read [Upgrade to a Patch Version of Astronomer](https://www.astronomer.io/docs/enterprise/v0.16/manage-astronomer/upgrade-astronomer-patch).
+> **Note:** Astronomer v0.16.9+ is required to run Astronomer Certified 1.10.12, and Astronomer v0.16.15+ is required to run Astronomer Certified 1.10.14. For instructions on how to upgrade to an Astronomer v0.16 patch version, read [Upgrade to a Patch Version of Astronomer](https://www.astronomer.io/docs/enterprise/v0.16/manage-astronomer/upgrade-astronomer-patch). As of Astronomer v0.23, compatibility of Astronomer Certified images is no longer dependent on a version of the platform.
```
Comment (jwitz) [inline enterprise/v0.16/09_resources/03_version-compatibility-reference.md:20]:
```suggestion
> **Note:** Astronomer v0.16.9+ is required to run Astronomer Certified 1.10.12, and Astronomer v0.16.15+ is required to run Astronomer Certified 1.10.14. For instructions on how to upgrade to an Astronomer v0.16 patch version, read [Upgrade to a Patch Version of Astronomer](https://www.astronomer.io/docs/enterprise/v0.16/manage-astronomer/upgrade-astronomer-patch). As of Astronomer v0.23, the platform is compatible with all versions of Astronomer Certified. 
```

This is what I thought this meant?

Comment (paolaperaza) [reply enterprise/v0.16/09_resources/03_version-compatibility-reference.md:20]: Ooo good suggestion, yes. Committing!

File: enterprise/next/09_resources/03_version-compatibility-reference.md [hunk 2]
```diff
@@ -12,23 +12,25 @@ It's worth noting that while the tables below reference the minimum compatible v
...
+| v0.23                | 1.16, 1.17, 1.18 | 3    | 0.13.5      | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.1 | 3.6, 3.7, 3.8  | 0.23           | 
 
-> **Note:** Astronomer v0.16.9+ is required to run Astronomer Certified 1.10.12, and Astronomer v0.16.15+ is required to run Astronomer Certified 1.10.14. For instructions on how to upgrade to an Astronomer v0.16 patch version, read [Upgrade to a Patch Version of Astronomer](https://www.astronomer.io/docs/enterprise/v0.16/manage-astronomer/upgrade-astronomer-patch).
+> **Note:** Astronomer v0.16.9+ is required to run Astronomer Certified 1.10.12, and Astronomer v0.16.15+ is required to run Astronomer Certified 1.10.14. For instructions on how to upgrade to an Astronomer v0.16 patch version, read [Upgrade to a Patch Version of Astronomer](https://www.astronomer.io/docs/enterprise/stable/manage-astronomer/upgrade-astronomer-patch). As of Astronomer v0.23, compatibility of Astronomer Certified images is no longer dependent on a version of the platform.
```
Comment (paolaperaza) [inline enterprise/next/09_resources/03_version-compatibility-reference.md:20]:
```suggestion
> **Note:** Astronomer v0.16.9+ is required to run Astronomer Certified 1.10.12, and Astronomer v0.16.15+ is required to run Astronomer Certified 1.10.14. For instructions on how to upgrade to an Astronomer v0.16 patch version, read [Upgrade to a Patch Version of Astronomer](https://www.astronomer.io/docs/enterprise/stable/manage-astronomer/upgrade-astronomer-patch). As of Astronomer v0.23, the platform is compatible with all versions of Astronomer Certified.
```

**Comment by paolaperaza** [comment]
Good call on those, thanks @jwitz - should be addressed!

**Comment by paolaperaza** [comment]
*Removed all references to 2.0.1 given @vishwas-astro's confirmation that we will now NOT be releasing AC 2.0.1.

**Comment by paolaperaza** [comment]
The CI job caught this link - https://packages.debian.org/stable/, but it works and is live for me so I'll go ahead and push forward.

**Pull request merged**